### ISSUE TITLE
OF-054 — Plan ML model readiness and first model strategy

### ISSUE THREAD

**Issue by lamadipen** [lamadipen/OpenFloodAI#108]
## Objective

Create a clear plan for when and how OpenFloodAI should start ML model work, while keeping the project camera-first.

In simple terms: before we train or plug in a model, we need to decide what data, labels, safety rules, and success checks are required. The first ML direction should support camera-based flood evidence, not replace the project with a general weather or flood-forecasting system.

## Background

The project is not ready for full ML model implementation yet. We first need better validation evidence, better frame sampling, and a small labelled dataset that we trust.

This issue is for planning only. It should not train a model yet.

We also received useful engineering input that OpenRiverCam is a strong open-source reference for camera-based water level, velocity, and discharge. Google Flood Forecasting / OpenHydroNet is useful for future basin-level forecasting research, but it is not the immediate camera MVP.

## Camera-First Boundary

Document this clearly:

```text
OpenFloodAI should focus first on a camera-based approach.

The core product is local camera evidence, human labels, machine comparison, and review.

Weather, rainfall, earthquake, hydrology, and forecasting data are useful future cross-checks, but they should not become the main MVP scope.
```

Plain-language version:

```text
First make the camera workflow reliable.
Later use rainfall, earthquake, and forecast data to double-check risk.
```

## Dependencies

This planning can start now, but final ML implementation should wait until at least these are clearer:

- #104: machine evidence samples frames across the labelled window
- #81: validation scorecard exists
- #106: review evidence images are meaningful
- #111: labelled data quality checklist exists

## Required Discussion Areas

Document a simple ML readiness plan covering:

- what labels we need first
- how many example videos or labelled windows we should collect before trying ML
- what counts as good training data
- what should stay local/private
- how OpenRiverCam may fit as upstream camera-monitoring inspiration or integration
- how Google Flood Forecasting may fit later as forecasting research, not immediate MVP implementation
- whether open-source segmentation models can help with assisted labelling
- whether Google ML, Vertex AI, or AutoML could help later
- what a first camera-based model should predict
- what metrics we should use to decide if the model is improving
- what safety boundaries must remain in place

## Tool And Model Comparison To Include

Include or link to a comparison table like this:

| Option | Best For | Not Best For | Recommended Use |
| --- | --- | --- | --- |
| OpenRiverCam / pyOpenRiverCam | Camera-based river water level, velocity, and discharge | General flood forecasting for an entire basin | Strong inspiration and possible future integration for upstream camera monitoring |
| Google Flood Forecasting / OpenHydroNet | Basin-level flood forecasting using hydrology, rainfall, streamflow, and catchment data | Local camera evidence review | Study later for forecasting ideas, not immediate MVP implementation |
| SAM 2 / YOLO segmentation | Helping mark water or river regions in images | Trusted flood warnings | Later for assisted labelling or water-region experiments |
| Classical computer vision | Cheap, explainable camera baselines like frame difference, optical flow, and reference-region change | Complex flood understanding by itself | Use now for the local MVP because it is simple, testable, and edge-friendly |
| FloodNet / satellite datasets | Learning flood-scene label ideas | Fixed river-camera monitoring | Research only for now |
| Vertex AI / Google Cloud Vision | Future cloud experiments and model management | Required edge operation or private-video workflow | Later only after privacy, cost, and data rules are clear |
| Rainfall / earthquake / external signals | Cross-checking whether a warning candidate makes sense | Main proof from the camera | Good future support signal, not the core MVP |

## Questions To Answer

At minimum, answer:

- Are we trying to detect flood/no flood, water rising/falling, water level, velocity, or a staged set of these?
- Do we train on full videos, frames, regions, or extracted measurements?
- Do we need to keep all videos/images, or only selected evidence and labels?
- How do we avoid training on private or unsafe data?
- Can OpenRiverCam outputs become optional upstream evidence records for OpenFloodAI?
- What should remain out of scope because it belongs to forecasting instead of camera evidence?
- What is the smallest useful camera-based ML experiment for this project?

## Safety Boundaries

This issue must not:

- train a production model
- upload private videos without a clear decision
- claim flood detection accuracy
- send alerts
- replace human review
- treat unverified labels as ground truth for production use
- make rainfall, earthquake, or forecast data the main MVP scope

## Expected Output

Create or update documentation with a short ML readiness plan.

Possible locations:

```text
docs/product/ml-readiness.md
docs/research/ml-model-options.md
```

or another existing docs location if the repository already has a better place.

## Acceptance Criteria

- [ ] A simple ML readiness plan exists in the docs.
- [ ] The plan explains that OpenFloodAI is camera-first for the current MVP.
- [ ] The plan explains what must be true before model training starts.
- [ ] The plan compares OpenRiverCam, Google Flood Forecasting, segmentation models, classical CV, datasets, cloud tools, and external signals.
- [ ] OpenRiverCam is treated as strong inspiration or possible integration for upstream camera monitoring.
- [ ] Google Flood Forecasting is treated as future forecasting research, not the immediate MVP path.
- [ ] Data privacy and safety boundaries are clear.
- [ ] The first possible camera-based ML experiment is described in plain language.
- [ ] No model training or production accuracy claim is added.

## Definition Of Done

The team can read the plan and know whether it is time to start ML work, what the first camera-based ML experiment should be, what role OpenRiverCam may play, and what must stay out of scope until later.

**Comment by lamadipen** [comment]
## MVP reference summary

Putting the MVP direction in one place so we can revisit it later.

### What MVP means right now

The MVP is a **local sample collection and validation system**. It is not a production flood warning system yet.

The main goal is simple:

> Give the project a video, choose the area to watch, run the validation, and compare what the machine saw with what a human saw.

For the first usable MVP, keep the flow small:

```text
One site
-> one site config
-> one video
-> one manifest row
-> one watched area
-> one validation run
-> one result folder
```

After that works clearly, we can repeat the same process with more videos.

### Main workflow

The UI should guide the user like this:

```text
Create site
-> Add video
-> Choose watched area
-> Save or repair manifest
-> Optional human label
-> Run validation
-> Review report, scorecard, records, and evidence images
-> Change watched area if needed
-> Run validation again
```

A user should also be able to work on one part directly. For example, they should be able to just change the watched area and rerun validation without starting from the beginning.

### Required to run validation

To run machine-only validation, we need:

- Site config
- Local video file
- Watched area / reference region
- Manifest row for tracking the video

Human label is **not required** for machine-only validation.

### Optional human label

Human label is only needed when we want to compare:

```text
What the machine saw
vs
What the human saw
```

So the system should support two valid states:

```text
Machine-only validation is ready
Machine + human comparison is ready
```

If human label is missing, validation should still run, but the report should clearly say that human comparison was skipped.

### Human evidence image

For now, human evidence image is **not required**.

A human can provide metadata/label notes without attaching an image. We closed the separate human evidence image story for now because the current system does not use that image in validation.

Later, if we need stronger review support, we can add evidence image upload again.

### UI rules

All UI work should start from:

```text
tools/openfloodai-home-ui.html
```

Please do not create a separate disconnected route or separate flow.

The MVP UI should feel like one connected workflow, but each step should still be usable on its own.

Important UI behavior:

- Use simple field names that normal users can understand.
- Dropdowns can show friendly labels, but saved data should still use stable machine values.
- Do not ask the user to manually check `Has human label` in the Add Video step. That should be derived from whether a human label record exists.
- Make it clear that labels are optional unless the user wants human comparison.
- Make the watched area easy to change for the same site/video.
- Make the Run Validation screen show readiness clearly before running.

### Pre-run readiness states

Before validation runs, the UI should clearly show one of these states:

```text
Ready for full comparison
Ready for machine-only validation
Blocked: required input is missing
```

Examples:

- If site config + video + watched area + manifest exist, but no human label exists: **Ready for machine-only validation**.
- If human label also exists: **Ready for full comparison**.
- If video or watched area is missing: **Blocked**.

### Safety boundaries

The MVP must stay conservative:

- Do not upload or publish private videos/images by default.
- `approved_for_repo` should stay false by default.
- Results are review evidence, not final flood proof.
- The system should not claim it can send real flood alerts yet.
- If the system cannot compare, that should be reported clearly. It should not be treated as success.

### Output expectation

Each validation run should preserve its own result folder.

Each run folder should keep related outputs together:

- Report
- Scorecard
- Records
- Review/evidence images
- Timestamp
- Final status

This lets us compare different runs later, especially when we change the watched area and rerun the same video.

### Known follow-up improvements

These are useful follow-ups after the main MVP flow is stable:

- Make sure default watched area does not pretend the site is fully configured unless the user selected or confirmed it.
- Make evidence images easier to find and open from the UI.
- Improve confusing report wording such as `Nones to Nones`; use plain text like `No usable frames`.
- Manifest repair should be a helper/recovery action. The normal Add Video flow should create the manifest row automatically.
- If users manually drop video files into folders, the UI should either repair them or clearly explain what is missing.

### What to prioritize next

Recommended next order:

1. Verify the one-site, one-video workflow end to end.
2. Confirm that changing watched area and rerunning creates a new preserved run folder.
3. Run a manual rehearsal with one known video.
4. Then run the same process with 5 sample videos.
5. Start building a small approved/public validation set.
6. Improve result explanations so a non-technical reviewer can understand what happened.
7. Define the ML readiness gate before starting model training.

### When to start ML/model training

We should not rush into model training yet.

Start ML planning when we have enough reviewed examples, such as:

- Enough labeled rising-water examples
- Enough falling-water examples
- Enough no-clear-change examples
- Enough bad camera/video examples
- A list of common failure cases
- A repeatable validation workflow that produces preserved results

A rough target could be 50-100 useful labeled examples before serious model training discussion.

For now, the camera-based validation workflow is the priority. Weather, rainfall, earthquake, and forecast data can be useful cross-checks later, but they are not the main MVP path.
